pyproject.toml

```diff
@@ -15,9 +17,10 @@ dependencies = [
     "networkx>=2.7",
     "numpy>=1.23",
     "scanpy>=1.10",
+    "leidenalg >=0.10.2",
 ]
 
-dynamic = ["version"]
+version = "0.2.0"
```

Comment:
I'm open to changing this, but why? 
If we specify the package version here, it should be removed from `__init__.py`.

Reply:
this might be nice tool to automate package versioning upon push to main: https://python-semantic-release.readthedocs.io/en/latest/

Reply:
Just for our overview: In a perfect world, after we push to `main` we want:

1. the version string to increase (minor/patch increase depending on commit messages)
2. the changelog to update (depending on commit messages)
3. the git commit be tagged with the version string
4. the git commit be used for a git release
5. the git release populated with the changelog content
6. the package updated on Pipy
7. the package updated on Bioconda

The Bioconda-bot should pick up the git release automatically. So if the version string is updated before the release that's done.

Reply:
> Here are some options to automate versioning (that do not require us to add files to the repo, like `versioneer`):
> 
> * https://python-semantic-release.readthedocs.io/en/latest/
> * https://dteslya.engineer/blog/2023/05/18/automating-python-project-releases-with-commitizen/#gitlab-ci

I like the second one, because it is more automated. The major disadvantage is that GitLab CI is different from GitHub Actions. Because we need to adjust and test it for GHA, I suggest we implement that in one of the next versions and not this one.

Reply:
> Just for our overview: In a perfect world, after we push to `main` we want:
> 
> 1. the version string to increase (minor/patch increase depending on commit messages, for example using this [commit message linter](https://github.com/marketplace/actions/semantic-pull-request) and [Python Semantic Release](https://python-semantic-release.readthedocs.io/en/latest/))
> 2. the changelog to update (depending on commit messages, also using Python Semantic Release)
> 3. the git commit be tagged with the version string
> 4. the git commit be used for a git release
> 5. the git release populated with the changelog content
> 6. the package updated on Pipy
> 7. the package updated on Bioconda (happens automatically with step 4)
> 8. (?) the package is uploaded to a container service (could be done with github actions [1](https://github.com/marketplace/actions/buildah-build) and [2](https://github.com/redhat-actions/push-to-registry))
> 
> The Bioconda-bot should pick up the git release automatically. So if the version string is updated before the release that's done.

That would indeed be ideal and something to aim for! The container service could either be the one from Conda itself or with a simple dockerfile we can use an action to push it to a docker registry ourselves.